### PR TITLE
Update citation information

### PR DESCRIPTION
**Pull request by femtobit**
With the release of our preprint, I suggest we update the citation information in our README and docs.

This PR also adds a `CITATIONS.bib` file, which should be [picked up by GitHub](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files#other-citation-files).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
 [![Release](https://img.shields.io/github/release/netket/netket.svg)](https://github.com/netket/netket/releases)
 [![Anaconda-Server Badge](https://anaconda.org/conda-forge/netket/badges/version.svg)](https://anaconda.org/conda-forge/netket)
-[![Paper](https://img.shields.io/badge/paper-SoftwareX%2010%2C%20100311%20(2019)-B31B1B)](https://doi.org/10.1016/j.softx.2019.100311)
+[![Paper (v3)](https://img.shields.io/badge/paper%20%28v3%29-arXiv%3A2112.10526-B31B1B)](https://arxiv.org/abs/2112.10526)
+[![Paper (v2)](https://img.shields.io/badge/paper%20%28v2%29-SoftwareX%2010%2C%20100311%20(2019)-B31B1B)](https://doi.org/10.1016/j.softx.2019.100311)
```
Comment (PhilipVinc) [inline README.md:10]: I'm not a fan of all those badges to be honest...

**Review by PhilipVinc — COMMENTED** [review]
You should also change the  link in the banner of the landing page .

I think we should decide what should be the citation policy we ask.
QuTiP asks that both v1 and v2 versions be cited. (They have no v3 paper ) see https://qutip.org/citing.html

I've wanted to also write in that page that if you use mpi you should also cite mpi4jax, otherwise people don't know.

Citations are cheap those days and we can ask to cite both. 
It's true that it is also a bit random, so... what do you think?

**Comment by gcarleo** [comment]
Since there is still significant overlap with v2 I would ask to cite both if possible... but I would only keep a badge with the last paper

**Comment by femtobit** [comment]
> Since there is still significant overlap with v2 I would ask to cite both if possible... but I would only keep a badge with the last paper

That's a fair point, especially considering that the current version still builds on contributions of people working on version 2.x, who aren't all co-authors of the new paper. I'll update the page.

File: README.md (hunk 1 above)
Comment (femtobit) [reply README.md:10]: Still, if we ask for citations of both papers, I think we should have both badges.

File: docs/citing.rst [hunk 2]
```diff
@@ -8,25 +8,67 @@
...
+    year={2021},
+    eprint={2112.10526},
+    archivePrefix={arXiv},
+    primaryClass={quant-ph}
+  }
+
+NetKet 2.x is described in `SoftwareX 10, 100311 (2019) <https://doi.org/10.1016/j.softx.2019.100311>`_.
+Despite many new features, NetKet 3 still builds on the foundations of the NetKet 2 era too. Therefore,
+we would appreciate it if you cite both papers in order to continue giving credit to all the authors of
+NetKet 2 as well.
```
Comment (PhilipVinc) [inline docs/citing.rst:35]: I think this should go on top?

I think we should have a text block on top with the links to both papers on top and explanation to our citation policy, plus the code block bibtex below

**Comment by PhilipVinc** [comment]
Unrelated to this PR, but I've long wanted to add a `nk.cite()` function that prints the bibtex file to the current folder plus a message, a bit like https://github.com/qutip/qutip/blob/master/qutip/cite.py .
Now that we have the Bibtex file in the repository it should be quite easy to do without pasting it in code.

**Review by PhilipVinc — APPROVED** [review]

**Comment by femtobit** [comment]
@gcarleo Do you also approve this version?

**Comment by gcarleo** [comment]
Yes just check a typo ressources

File: docs/citing.rst [hunk 3]
```diff
@@ -8,25 +8,70 @@
 
       Sustain NetKet's developers
 
+NetKet
+------
 
-If you find the NetKet project useful for your research, please consider citing `our paper <https://www.sciencedirect.com/science/article/pii/S2352711019300974>`_ using this ready-made BibTeX entry
+If you find the NetKet project useful for your research, please consider citing our relevant software papers
+in order to help us to continue devoting time and ressources to NetKet development.
```
Comment (femtobit) [inline docs/citing.rst:15]:
```suggestion
If you find the NetKet project useful for your research, please consider citing our relevant software papers
in order to help us to continue devoting time and resources to NetKet development.
```

**Pull request merged**